accepted/2024/[TS]PMUI-Readme-rendering.md

```diff
@@ -30,19 +30,104 @@ When no README is available we will display a messsage in the README section.
+    "resources": [
+        ...,
+        {
+            "@id": "https://apidev.nugettest.org/v3/registration5-gz-semver2/",
+            "@type": "RegistrationsBaseUrl/6.12.0"
+        },
+        ...
+    ]
+}
+```
+###### Example Response
+```json
+{
+    "@id": "https://apidev.nugettest.org/v3/registration5-gz-semver2/newtonsoft.json/index.json",
+    "items": [
+        {
+            ...,
+            "items": [ 
+                ...,
+                {
+                    "@id": "https://apidev.nugettest.org/v3/registration5-gz-semver2/newtonsoft.json/13.0.3.json",
+                    "@type": "Package",
+                    "catalogEntry": {
+                        "@id": "https://apidev.nugettest.org/v3/catalog0/data/2024.08.19.16.55.59/newtonsoft.json.13.0.3.json",
+                        "@type": "PackageDetails",
+                        "authors": "James Newton-King",
+                        "dependencyGroups": [ ... ],
+                        "description": "Json.NET is a popular high-performance JSON framework for .NET",
+                        "iconUrl": "https://apidev.nugettest.org/v3-flatcontainer/newtonsoft.json/13.0.3/icon",
+                        "id": "Newtonsoft.Json",
+                        "language": "",
+                        "licenseExpression": "MIT",
+                        "licenseUrl": "https://dev.nugettest.org/packages/Newtonsoft.Json/13.0.3/license",
+                        "readmeUrl": "https://dev.nugettest.org/packages/Newtonsoft.Json/13.0.3#show-readme-container",
+                        //New field
+                        "rawReadmeUrl": "https://apidev.nugettest.org/v3/flatcontainer/newtonsoft.json/13.0.3/readme",
+                        "listed": true,
+                        "minClientVersion": "2.12",
+                        "packageContent": "https://apidev.nugettest.org/v3-flatcontainer/newtonsoft.json/13.0.3/newtonsoft.json.13.0.3.nupkg",
+                        "projectUrl": "https://www.newtonsoft.com/json",
+                        "published": "2023-04-25T14:48:53.817+00:00",
+                        "requireLicenseAcceptance": false,
+                        "summary": "",
+                        "tags": [
+                            "json"
+                        ],
+                        "title": "Json.NET",
+                        "version": "13.0.3"
+                    },
+                    "packageContent": "https://apidev.nugettest.org/v3-flatcontainer/newtonsoft.json/13.0.3/newtonsoft.json.13.0.3.nupkg",
+                    "registration": "https://apidev.nugettest.org/v3/registration5-gz-semver2/newtonsoft.json/index.json"
+                },
+                ...
+            ],
+            ...
+        }
+    ],
+    ...
+}
+```
+##### PackageBaseAddress/6.12.0
 
 A new version of the [package content](https://learn.microsoft.com/en-us/nuget/api/package-base-address-resource) resource type which will include a url definition for downloading the README.
```

Comment:
Pedantic: This example below is not showing the URL to download the readme, it's showing a base URL, for which a URL has to be generated.  It also doesn't tell us what the generated URL will be.  See [the docs](https://learn.microsoft.com/en-us/nuget/api/package-base-address-resource) for examples how they explain to both service and client implementors what the expected URL will be.

Guessing from the registration example's rawReadMeUrl, I assume the proposed filename will be `readme`.  Consider the two documented resources both have file extensions, and embedded readmes only support markdown, should it be `readme.md`?

Reply:
I can update the doc, as for the URI. Users can pick whatever name they want for the README. If we set the api {@id}/{LOWER_ID}/{LOWER_VERSION}/readme.md it implies like are specifically looking for that file. As opposed to /readme which leaves it open to whatever readme is available for that package.

Reply:
yes, the doc needs to explain how server implementors tell client what filename to use (what pattern client will use to generate the URl that will be requested)